File: round1_full_pack_optionA_solutions/M2_Longest_Substring_No_Repeat/Solution/Solution.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        if (!sc.hasNext()) {
            System.out.println(0);
            return;
        }

        String s = sc.next();

        int[] last = new int[256];
        Arrays.fill(last, -1);

        int l = 0, best = 0;

        for (int r = 0; r < s.length(); r++) {
            char c = s.charAt(r);

            if (last[c] >= l)
                l = last[c] + 1;

            last[c] = r;

            best = Math.max(best, r - l + 1);
        }

        System.out.println(best);
    }
}

File: round1_full_pack_optionA_solutions/M2_Longest_Substring_No_Repeat/Solution/Solution.py
import sys

s = sys.stdin.readline().strip()

if not s:
    print(0)
    sys.exit()

last = [-1] * 256
l = 0
best = 0

for r in range(len(s)):
    c = ord(s[r])

    if last[c] >= l:
        l = last[c] + 1

    last[c] = r
    best = max(best, r - l + 1)

print(best)
